From 743bb45e49ac30884ec92a7ded21181bea74f822 Mon Sep 17 00:00:00 2001
From: Julia Cartwright <julia@ni.com>
Date: Thu, 16 Feb 2017 16:57:21 -0600
Subject: [PATCH 125/134] trace-cmd: Fixup more bogus symbol resolutions

Commit ce1cbc51c ("trace-cmd: Leave out absolute addresses to fix bogus
symbol resolutions") fixed up bogus symbol resolution for 'A' type
symbols used on x86_64 for percpu offsets.

However, this check did not cover per-cpu variables used in modules,
which are generated as 'local' absolute symbols, and therefore appear as 'a' in
/proc/kallsyms.  Fix this.

Link: http://lkml.kernel.org/r/20170216225721.13087-1-julia@ni.com

Cc: Jan Kiszka <jan.kiszka@siemens.com>
Signed-off-by: Julia Cartwright <julia@ni.com>
Signed-off-by: Steven Rostedt (VMware) <rostedt@goodmis.org>
---
 trace-util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/trace-util.c b/trace-util.c
index 4073c790d5ae..fbf8cea26dda 100644
--- a/trace-util.c
+++ b/trace-util.c
@@ -459,7 +459,7 @@ void tracecmd_parse_proc_kallsyms(struct pevent *pevent,
 		 *  - arm arch that adds a lot of bogus '$a' functions
 		 *  - x86-64 that reports per-cpu variable offsets as absolute
 		 */
-		if (func[0] != '$' && ch != 'A')
+		if (func[0] != '$' && ch != 'A' && ch != 'a')
 			pevent_register_function(pevent, func, addr, mod);
 		free(func);
 		free(mod);
-- 
2.5.5

